pallets/gear/src/schedule.rs

```diff
@@ -660,107 +658,72 @@ impl<T: Config> From<MemoryWeights<T>> for LazyPagesCosts {
     }
 }
 
-macro_rules! replace_token {
-    ($_in:tt $replacement:tt) => {
-        $replacement
-    };
-}
-
-macro_rules! call_zero {
-    ($name:ident, $( $arg:expr ),*) => {
-        <T as Config>::WeightInfo::$name($( replace_token!($arg 0) ),*)
-    };
-}
-
-macro_rules! cost_args {
-    ($name:ident, $( $arg: expr ),+) => {
-        (<T as Config>::WeightInfo::$name($( $arg ),+).saturating_sub(call_zero!($name, $( $arg ),+))).ref_time()
-    }
-}
-
-macro_rules! cost_batched_args {
-    ($name:ident, $( $arg: expr ),+) => {
-        cost_args!($name, $( $arg ),+) / u64::from(API_BENCHMARK_BATCH_SIZE)
-    }
-}
-
-macro_rules! cost_instr_no_params_with_batch_size {
-    ($name:ident, $batch_size:expr) => {
-        (cost_args!($name, 1) / u64::from($batch_size)) as u32
-    };
-}
-
-macro_rules! cost_instr_with_batch_size {
-    ($name:ident, $num_params:expr, $batch_size:expr) => {
-        cost_instr_no_params_with_batch_size!($name, $batch_size).saturating_sub(
-            (cost_instr_no_params_with_batch_size!(instr_i64const, $batch_size) / 2)
-                .saturating_mul($num_params),
-        )
-    };
+#[inline]
+fn cost(w: fn(u32) -> Weight) -> Weight {
+    Weight::from_parts(w(1).saturating_sub(w(0)).ref_time(), 0)
 }
 
-macro_rules! cost_instr {
-    ($name:ident, $num_params:expr) => {
-        cost_instr_with_batch_size!($name, $num_params, INSTR_BENCHMARK_BATCH_SIZE)
-    };
+#[inline]
+fn cost_byte(w: fn(u32) -> Weight) -> Weight {
+    Weight::from_parts(cost(w).ref_time() / 1024, 0)
 }
 
-macro_rules! cost_byte_args {
-    ($name:ident, $( $arg: expr ),+) => {
-        cost_args!($name, $( $arg ),+) / 1024
-    }
+#[inline]
+fn cost_batched(w: fn(u32) -> Weight) -> Weight {
+    Weight::from_parts(cost(w).ref_time() / u64::from(API_BENCHMARK_BATCH_SIZE), 0)
 }
 
-macro_rules! cost_byte_batched_args {
-    ($name:ident, $( $arg: expr ),+) => {
-        cost_batched_args!($name, $( $arg ),+) / 1024
-    }
+#[inline]
+fn cost_byte_batched(w: fn(u32) -> Weight) -> Weight {
+    Weight::from_parts(cost_batched(w).ref_time() / 1024, 0)
 }
 
-macro_rules! cost {
-    ($name:ident) => {
-        cost_args!($name, 1)
-    };
+#[inline]
+fn cost_byte_batched_args(w: fn(u32, u32) -> Weight, arg1: u32, arg2: u32) -> Weight {
```

Comment:
Currently, Rust does not have functions that can accept a variable number of arguments, and in such cases, it usually makes sense to use macros. However, we have only one such case here, so it seems fine to handle it manually.